scripts/packaging/publishupmpackages.ps1

```diff
@@ -3,36 +3,51 @@
     Publishes the Mixed Reality Toolkit Unity Package Manager (UPM) packacges.
 .DESCRIPTION
     Publishes UPM packages for the Mixed Reality Toolkit.
+.PARAMETER Publish
+    Indicates whether or not to publish the packages.
 .PARAMETER PackageDirectory
-    Where should we find the packages to upload? Defaults to ".\artifacts"
-.PARAMETER ProjectRoot
-    The root folder of the project.
+    Where should we find the packages to upload? Defaults to ".\artifacts\upm"
+.PARAMETER RegistryPath
+    To which registry should the packages be uploaded?
+
 #>
 param(
-    [string]$PackageDirectory = ".\artifacts\upm",
-    [string]$ProjectRoot
+    [string]$Publish = "false",

```

Comment:
Apologies for not leaving this comment earlier - I think that this parameter should be removed because it's a required boolean param.

It basically adds no value - if it isn't called with Publish = true, then this immediately dies. It doesn't really make sense to have a general command line tool that is like:

RunCommandLineTool -param1=path\to\thing

And it just fails unless you call it with a parameter that does absolutely nothing except actually let it run:

RunCommandLineTool -param1=path\to\thing -necessaryButNonValueBasedParameter

Reply:
this is what allows the script to differentiate between the two pipeline versions (the one that publishes every build to the test server and the one does not publish to the public server)

Reply:
let me try one other thing, if it works we're golden. if not, let's use the above for now.